### PR TITLE
Create npm-audit-fix

### PR DESCRIPTION
**Pull request by szaimen**
Idea: you can simply use the workflow_dispatch to run npm audit fix for all needed branches in question without having to deal with it locally.

### CODE REVIEW

**Comment by szaimen** [comment]
> Unpopular opinion: set up dependabot or renovate and configure it correctly

I rather have both: renovate/dependabot configured correctly and having the option to trigger this workflow

File: workflow-templates/npm-audit-fix.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+      - name: Install dependencies & fix npm audit
+        run: |
+          npm audit fix
+
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@18f7dc018cc2cd597073088f7c7591b9d1c02672 # v3
+        with:
+          token: ${{ secrets.COMMAND_BOT_PAT }}
+          commit-message: "chore(dev-deps): fix npm audit"
```
[the remark this answers is not in the document]
Comment (szaimen) [reply workflow-templates/npm-audit-fix.yml]: done

**Comment by nickvergessen** [comment]
Should it check for js/ or we have 2 different versions and then automatically commit compiled when available, so we save a roundtrip of CI?

**Comment by skjnldsv** [comment]
> Unpopular opinion: set up dependabot or renovate and configure it correctly

We have dependabot everywhere, and yet I've seen cases where an npm audit was necessary :thinking:

**Comment by nickvergessen** [comment]
> We have dependabot everywhere

In most repos it's only set up for master/main and not for any stable branches, e.g. I just did it for notifications 2 weeks ago, after how many years of JS in that repo? :P
https://github.com/nextcloud/notifications/commit/de0f2da68582d1f8a6c61866adc268e71f001f93

And for the ones we do, we most of the time limit it to patch updates which sometimes might not be enough. But it should still be done to give a better state overall.

Anyway, I'm fine with doing both.

> you mean auomtatic compile after the command and commit the changes?

Yes

**Comment by szaimen** [comment]
> > you mean auomtatic compile after the command and commit the changes?
> 
> Yes

done

**Review by Pytal — APPROVED** [review]

**Comment by skjnldsv** [comment]
> And for the ones we do, we most of the time limit it to patch updates which sometimes might not be enough. But it should still be done to give a better state overall.

Dependabot update and security fixes are two different things.
Security dependabot bumps do NOT rely on your `dependabot.yml` file config :wink:

**Pull request merged**